### PR TITLE
chore(deps): use replace appdirs by platformdirs

### LINKED ISSUE #661: appdirs is deprecated
See their README at https://github.com/ActiveState/appdirs. They recommend to move to https://github.com/platformdirs/platformdirs. I might do it in my free time.

### PR DESCRIPTION
**Pull request by deronnax**
closes #661

### CODE REVIEW

File: changelog.d/20240125_163926_mathieu.dupuy_replace_appdirs_by_platform_dir.md
Comment (agateau-gg) [inline changelog.d/20240125_163926_mathieu.dupuy_replace_appdirs_by_platform_dir.md:1]: The right way to please the CI for internal changes is to add the skip-changelog label, no need for a dummy changelog entry :)

File: setup.cfg [hunk 1]
```diff
@@ -36,7 +36,7 @@ python_requires = >=3.8
 packages = find:
 zip_safe = True
 install_requires =
-    appdirs>=1.4.4,<1.5.0
+    platformdirs>=3.0.0
```
Comment (agateau-gg) [inline setup.cfg:39]: "Pin all the things!"

```suggestion
    platformdirs>=3.0.0,<3.1.0
```

**Review by agateau-gg — CHANGES_REQUESTED** [review]

File: changelog.d/20240125_163926_mathieu.dupuy_replace_appdirs_by_platform_dir.md
Comment (deronnax) [reply changelog.d/20240125_163926_mathieu.dupuy_replace_appdirs_by_platform_dir.md:1]: arg, my changelog entry got lost in the process. But it looks like it allowed us to spot a hole in the net.

File: setup.cfg (hunk 1 above)
Comment (deronnax) [reply setup.cfg]: done

**Review by agateau-gg — APPROVED** [review]

**Pull request merged**